docs/docs/reference/experimental/cc.md

```diff
@@ -0,0 +1,716 @@
+  | - Wrapping this piece of code with a `try` block that catches LimitExceeded
+```
+`CanThrow` capabilities are required by `throw` expressions and are created
+by `try` expressions. For instance, the expression
+```scala
+try xs.map(f).sum
+catch case ex: LimitExceeded => -1
+```
+would be expanded by the compiler to something like the following:
+```scala
+try
+  erased given ctl: CanThrow[LimitExceeded] = compiletime.erasedValue
+  xs.map(f).sum
+catch case ex: LimitExceeded => -1
+```
+(The `ctl` capability is only used for type checking but need not show up in the generated code, so it can be declared as
+erased.)
+
+As with other capability based schemes, one needs to guard against capabilities
+that are captured in results. For instance, here is a problematic use case:
+```scala
+def escaped(xs: Double*): (() => Double) throws LimitExceeded =
+  try () => xs.map(f).sum
+  catch case ex: LimitExceeded => () => -1
+val crasher = escaped(1, 2, 10e+11)
+crasher()
+```
+This code needs to be rejected since otherwise the call to `later()` would cause
+an unhandled `LimitExceeded` exception to be thrown.
+
+Under `-Ycc`, the code is indeed rejected
+```
+14 |  try () => xs.map(f).sum
+   |  ^
+   |The expression's type {*} () -> Double is not allowed to capture the root capability `*`.
+   |This usually means that a capability persists longer than its allowed lifetime.
+15 |  catch case ex: LimitExceeded => () => -1
+```
+To integrate exception and capture checking, only two changes are needed:
+
+ - `CanThrow` is declared as a `@capability` class, so all references to `CanThrow` instances are tracked.
+ - Escape checking is extended to `try` expressions. The result type of a `try` is not allowed to
+   capture the universal capability.
+
+
+
+## A Larger Example
+
+As a larger example, we present an implementation of lazy lists and some use cases. For simplicity,
+our lists are lazy only in their tail part. This corresponds to what the Scala-2 type `Stream` did, whereas Scala 3's `LazyList` type computes strictly less since it is also lazy in the first argument.
+
+Here is the base trait `LzyList` for our version of lazy lists:
+```scala
+trait LzyList[+A]:
+  def isEmpty: Boolean
+  def head: A
+  def tail: {this} LzyList[A]
+```
+Note that `tail` carries a capture annotation. It says that the tail of a lazy list can
+potentially capture the same references as the lazy list as a whole.
+
+The empty case of a `LzyList` is written as usual:
+```scala
+object LzyNil extends LzyList[Nothing]:
+  def isEmpty = true
+  def head = ???
+  def tail = ???
+```
+Here is a formulation of the class for lazy cons nodes:
+```scala
+import scala.compiletime.uninitialized
+
+final class LzyCons[+A](hd: A, tl: () => {*} LzyList[A]) extends LzyList[A]:
+  private var forced = false
+  private var cache: {this} LzyList[A] = uninitialized
+  private def force =
+    if !forced then { cache = tl(); forced = true }
```

Comment:
Does this mean that `{*} LzyList[A] <: {this} LzyList[A]` ? How does this follow exactly?

Reply:
No, it's more subtle. See https://github.com/lampepfl/dotty/discussions/14387

I still need to work that in the discussion here somehow.